app/src/test/java/org/mozilla/fenix/ext/BookmarkNodeTest.kt

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ext
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.TestApplication
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import mozilla.components.concept.storage.BookmarkNode
+import mozilla.components.concept.storage.BookmarkNodeType
+
+@RunWith(RobolectricTestRunner::class)
+@Config(application = TestApplication::class)
+class BookmarkNodeTest {
+    val nodeType = BookmarkNodeType.ITEM
+    val bookMarkChild1 = newBookMarkNode("Child 1", 1, null)
+    val bookMarkChild2 = newBookMarkNode("Child 2", 2, null)
+    val bookMarkChild3 = newBookMarkNode("Child 3", 3, null)
+    val allChildren = listOf(bookMarkChild1, bookMarkChild2)
+    val subsetToSubtract = setOf(bookMarkChild1)
+    val expectedRemainingSubset = listOf(bookMarkChild2)
```

Comment:
nit: this is only used in a single test so it's definition should be moved into the test.

In general, you should try to colocate data creation to the test that uses it because, if you need to update the data in a way that will change depending on the test, it's much more complicated to do so if the data is defined globally. For example:
```
// Used by two tests
val bmNode = BookmarkNode(Type.WHATEVER, children = null)

// But the BookmarkNode type changes such that we need two values:
val bmNode1 = BookmarkNode(Type.NO_CHILDREN, children = null)
val bmNode2 = BookmarkNode(Type.NO_CHILDREN_WITH_PARENT, children = null)
```

This is a weird contrived example but the general idea is that the dependencies between the data and the tests that uses it is unclear so it's hard to change the data, if it needs to be updated for whatever reason. It's better to colocate the data. /ramble 😁

Reply:
Just a note that this was not addressed.